lochness/rpms/__init__.py

```diff
@@ -235,20 +250,26 @@ def get_subject_data(all_df_dict: Dict[str, pd.DataFrame],
 
     subject_df_dict = {}
     for measure, measure_df in all_df_dict.items():
+        if id_colname not in measure_df.columns:
+            continue
+
         measure_df[id_colname] = measure_df[id_colname].astype(str)
         subject_df = measure_df[measure_df[id_colname] == subject.id]
 
-        # Keep the most recent row for each visit
-        for unique_visit, table in subject_df.groupby('visit'):
-            if len(table) == 1:
-                pass
-            else:
-                most_recent_row_index = pd.to_datetime(
-                        table['LastModifiedDate']).idxmax()
-                non_recent_row_index = [x for x in table.index
-                         if x != most_recent_row_index]
-                print(f'RPMS export has duplicated rows for {measure}')
-                subject_df.drop(non_recent_row_index, inplace=True)
+        # RPMS should overwrite the row whenever there is an update in any 
+        # field of the visit. The snippet below is a safety measure, to store
+        # most recent visit row for each visit
```

Comment:
A continuation of the thought I shared on MS Teams, do RPMS keep all rows of all visits until that time?